Assets/Source/Components/Enemy/Base/EnemyAIBase.cs

```diff
@@ -0,0 +1,78 @@
+﻿using UnityEngine;
+using Assets.Source.Components.Base;
+using Assets.Source.Constants;
+using Assets.Source.Components.Actor;
+using Assets.Source.Extensions;
+
+namespace Assets.Source.Components.Enemy.Base
+{
+    public abstract class EnemyAIBase : ComponentBase
```

Comment:
I actually think the logic in this class is good.  Its pretty lightweight and everything in here should be shared logic.